pkg/tarfs/tarfs_test.go

```diff
@@ -460,6 +460,83 @@ func TestSymlinks(t *testing.T) {
 	}))
 }
 
+func TestReadFile(t *testing.T) {
+	type tarfsFile struct {
+		Header tar.Header
+		Data   []byte
+	}
+
+	tmp := t.TempDir()
+	setupAndRun := func(openErr bool, tarfsFiles []tarfsFile, chk func(*testing.T, *FS)) {
+		t.Helper()
+		// This is a perfect candidate for using test.GenerateFixture, but
+		// creates an import cycle.
+		f, err := os.Create(filepath.Join(tmp, path.Base(t.Name())))
```

Comment:
I think I used `path` throughout and swapped specific uses because `io/fs` mandates forward-slash paths and go just handles forward-slash paths, so it's moot unless the path could be absolute.